### DISCUSSION TITLE
Data Collection: Why?

### DISCUSSION THREAD

**Discussion by gh-PonyM** [AntonOsika/gpt-engineer#935] [Q&A]
Seriously, can you explain to me why you collect data from this project that uses AI services from a company that literally invests billions of dollars? How exactly you want to improve this system by collecting user input compared to the company you rely on?

**Comment by ErikBjare** [comment]
I haven't looked at the collected data, but there are lots of obvious things it enables (tracking failure modes, fine-tuning, evals). This is pretty standard practice for projects in this space, although less common for open-source projects.

Why the tone?

**Reply by gh-PonyM** [reply]
Sorry for the tone but maybe let me explain the issue I have with it: I a recent talk about AI strategy lead by ETH and EPLF in Switzerland, one researcher raised the question: Why does nobody discusses the fact, that companies like OpenAI and Micrsoft (invested with more than 10Billion $) can make profit out of public data (fed into AI)? You might ask: Yes but it's public information. Yes - but consider some cases here: For example ClearView AI is not allowed to scrape image from people inside the EU to train their algos to recognize people - even thought it's public information, the DGSVO guarantees some right concerning usage of public data. 

Now, I am not aware of any other MIT licensed open-source project that collects data about it's usage in a direct manner for a hand of elected people. For now, I have not looked into the source code what is transmitted to your backend. But I am aware that this project can go private any time and use that data using the mentioned license. So as you said - it's rather unusual for an open-source project, less so for AI related things, I agree. Another way to got is to make this information public to enable others to improve this as well or at least show how it's used to improve this very project. Or people, just fork and remove that code. But anyway, my rant goes much broader from the principle as I have had a detailed look into the code.
